src/Compilers/CSharp/Test/Semantic/Semantics/FunctionPointerTests.cs

```diff
@@ -3384,12 +3384,15 @@ .maxstack  2
 
             VerifyOperationTreeForNode(comp, model, isPatterns[0], expectedOperationTree: @"
 IIsPatternOperation (OperationKind.IsPattern, Type: System.Boolean) (Syntax: 'ptr is null')
-  Value: 
+  Value:
     ILocalReferenceOperation: ptr (OperationKind.LocalReference, Type: delegate*<System.Void>) (Syntax: 'ptr')
-  Pattern: 
+  Pattern:
     IConstantPatternOperation (OperationKind.ConstantPattern, Type: null) (Syntax: 'null') (InputType: delegate*<System.Void>, NarrowedType: delegate*<System.Void>)
-      Value: 
-        ILiteralOperation (OperationKind.Literal, Type: null, Constant: null) (Syntax: 'null')
+      Value:
+        IConversionOperation (TryCast: False, Unchecked) (OperationKind.Conversion, Type: delegate*<System.Void>, IsImplicit) (Syntax: 'null')
```

Comment:
>IConversionOperation (TryCast: False, Unchecked) (OperationKind.Conversion, Type: delegate*<System.Void>, IsImplicit) (Syntax: 'null') [](http://example.com/codeflow?start=8&length=134)

It feels rather strange that an operand of a constant pattern doesn't have a constant value in a success case. Instead of changing the shape of the bound tree, could we adjust consumers to deal with bound null instead?

Reply:
https://github.com/dotnet/roslyn/pull/70113